### PR TITLE
feat: expose recent tinymist logs to tools and AI agents

### PR DESCRIPTION
**Pull request by flol3622**
## Summary

This adds programmatic access to recent TinyMist VS Code extension logs.

Today, `tinymist.showLog` only opens the VS Code Output panel. That works for humans, but not for tooling. In practice, AI agents and other automation can often inspect diagnostics and command results, but cannot read raw Output panel contents. That makes TinyMist failures harder to diagnose unless the user manually copies the log text.

This change adds a bounded in-memory buffer around the existing `Tinymist Typst` output channel and exposes a hidden `tinymist.getLogText` command that returns recent log text, optionally truncated with `maxChars`.

## Why

The main motivation is to make TinyMist logs accessible to AI agents and other tools.

Typical agent workflows in VS Code can read files, diagnostics, and command results, but they usually cannot inspect the Output panel directly because VS Code output channels are effectively write-only to external tooling. That creates a blind spot for compile and runtime failures that TinyMist logs but does not surface through normal diagnostics.

By exposing recent logs through a command:

- AI agents can inspect TinyMist output without requiring the user to copy-paste logs.
- Other tooling can programmatically retrieve recent log text for debugging or integration.
- Existing UI behavior stays unchanged for human users.

## What changed

- Added a buffered output channel wrapper in `editors/vscode/src/lsp.ts`.
- Added `LanguageState.getLogText(maxChars?)` in `editors/vscode/src/lsp.ts`.
- Registered a hidden `tinymist.getLogText` command in `editors/vscode/src/extension.shared.ts`.
- Added an e2e test in `editors/vscode/src/test/e2e/simple-docs.test.ts`.

## Behavior

- `tinymist.showLog` still opens the Output panel.
- `tinymist.getLogText` returns the buffered recent log text.
- Passing `maxChars` returns only the most recent tail.
- Clearing the output channel clears the in-memory buffer as well.

## Testing

- Added e2e coverage for reading recent log text and truncation behavior.
- The touched files are clean under editor diagnostics.
- A full extension typecheck in the local clone still reported one unrelated existing error in `src/lsp.code-action.ts:49`.

## Notes

This is intentionally implemented as a hidden command rather than a new visible UI command because the primary consumer is tooling rather than end users.

### CODE REVIEW

**Comment by flol3622** [comment]
Updated the branch to local HEAD `16429c74`.

What changed in this update:
- Fixed the zero-length edge case in `BufferedOutputChannel.getText`, so `maxChars: 0` now returns an empty string instead of the full buffer.
- Fixed the zero-capacity edge case in `BufferedOutputChannel.push`, so a `maxChars <= 0` buffer cannot grow unexpectedly.
- Added test coverage for the `maxChars: 0` case.
- Republished the branch from the exact local git history, which also restores the full `lsp.ts` content after the earlier branch upload mismatch.

The open PR should now reflect the intended code rather than the earlier incomplete remote state.

File: editors/vscode/src/lsp.ts [hunk 1]
```diff
@@ -868,43 +960,3 @@ export interface SymbolInfo {
   kind: string;
   children: SymbolInfo[];
 }
```
[the remark this answers is not in the document]
Comment (flol3622) [reply editors/vscode/src/lsp.ts:966]: This was an artifact of the initial branch upload where the file was accidentally truncated mid-way through. The force-push in the latest revision restores the full file — all of `isCodeActionWithoutEditsAndCommands`, `FsWatchRequest`, `FileResult`, `FileChange`, `FsChangeParams`, and `fsChange` are present and unchanged from upstream. This comment is now outdated.

File: editors/vscode/src/extension.shared.ts [hunk 2]
```diff
@@ -239,25 +242,15 @@ function makeExecCommand(
...
-        });
+      const exited = await new Promise<{ code: number | null; signal: string | null }>((resolve) => {
+        proc.on("close", (code, signal) => resolve({ code, signal }));
       });
+
+      return {
+        stdout: Buffer.concat(capturedStdout),
+        stderr: Buffer.concat(capturedStderr),
+        ...exited,
+      };
```
[the remark this answers is not in the document]
Comment (flol3622) [reply editors/vscode/src/extension.shared.ts]: Same root cause as the `lsp.ts` truncation above — the initial upload cut `extension.shared.ts` short, losing the bottom of `makeExecCommand`. The force-pushed branch has the original implementation intact, including `proc.on("error", reject)` and `code || 0` for `null`-safety. This comment is now outdated.

File: editors/vscode/src/lsp.ts [hunk 3]
```diff
@@ -93,11 +93,99 @@ interface JumpInfo {
...
+    this.channel.dispose();
+  }
+
+  getText(maxChars?: number): string {
+    if (maxChars === undefined || maxChars < 0 || this.buffer.length <= maxChars) {
+      return this.buffer;
+    }
+
+    return this.buffer.slice(-maxChars);
+  }
```
[the remark this answers is not in the document]
Comment (flol3622) [reply editors/vscode/src/lsp.ts:174]: Fixed in commit `16429c74`. `getText` now has an explicit `if (maxChars === 0) return "";` guard as the first branch, before any `slice` call, so requesting 0 chars returns `""` rather than the full buffer. Test coverage added: `ctx.expect(none).to.equal("")`.

File: editors/vscode/src/lsp.ts [hunk 4]
```diff
@@ -93,11 +93,99 @@ interface JumpInfo {
...
+  private push(value: string): void {
+    if (!value) {
+      return;
+    }
+
+    this.buffer += value;
+    if (this.buffer.length > this.maxChars) {
+      this.buffer = this.buffer.slice(-this.maxChars);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (flol3622) [reply editors/vscode/src/lsp.ts:185]: Fixed in commit `16429c74`. `push` now opens with `if (!value || this.maxChars <= 0) return;`, which prevents any buffer accumulation when the channel is constructed with a zero capacity. The `slice(-0)` truncation path is therefore never reached in that case.
